shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/data/FirebaseOrdersDataSource.kt

```diff
@@ -5,16 +5,36 @@ import com.vml.tutorial.plantshop.profile.orders.domain.OrderItem
 import com.vml.tutorial.plantshop.profile.orders.domain.OrdersDataSource
 import dev.gitlive.firebase.Firebase
 import dev.gitlive.firebase.firestore.CollectionReference
+import dev.gitlive.firebase.firestore.Direction
 import dev.gitlive.firebase.firestore.FirebaseFirestore
 import dev.gitlive.firebase.firestore.firestore
+import dev.gitlive.firebase.firestore.orderBy
 import dev.gitlive.firebase.firestore.where
 
 class FirebaseOrdersDataSource(
     private val storage: FirebaseFirestore = Firebase.firestore,
-    private val ordersCollection: CollectionReference = storage.collection(ORDERS)
+    private val ordersCollection: CollectionReference = storage.collection(ORDERS_COLLECTION)
 ): OrdersDataSource {
-    override suspend fun fetchOrders(userId: String): List<OrderItem> {
-        return ordersCollection.where(USER_ID, userId).getCollection<OrderItem>()
+    override suspend fun fetchOrders(userId: String, queries: Map<String, String>): List<OrderItem> {
+        var query = ordersCollection.where(Fields.USER_ID, userId)
```

Comment:
Could we use method chaining using apply?

Reply:
Optimized it thanks :)